symfony/plugins/orangehrmAdminPlugin/client/jobCategory/views/JobCategoryList.vue

```diff
@@ -0,0 +1,175 @@
+            <oxd-button
+              label="Delete Selected"
+              displayType="label-danger"
+              @click="onClickDeleteSelected"
+              class="orangehrm-horizontal-margin"
+            />
+          </div>
+          <oxd-text tag="span" v-else> {{ total }} Job Category Found</oxd-text>
+        </div>
+      </div>
+      <div class="orangehrm-container">
+        <oxd-card-table
+          ref="dTable"
+          :headers="headers"
+          :items="items"
+          :selectable="true"
+          v-model:selected="checkedItems"
+          rowDecorator="oxd-table-decorator-card"
+        />
+      </div>
+      <div class="orangehrm-bottom-container">
+        <oxd-pagination
+          v-if="showPaginator"
+          :length="pages"
+          v-model:current="currentPage"
+        />
+      </div>
+    </div>
+
+    <delete-confirmation ref="deleteDialog"></delete-confirmation>
+  </div>
+</template>
+
+<script>
+import DeleteConfirmationDialog from "@orangehrm/components/dialogs/DeleteConfirmationDialog";
+
+export default {
+  data() {
+    return {
+      headers: [
+        { name: "name", title: "Job Category", style: { "flex-basis": "80%" } },
+        {
+          name: "actions",
+          title: "Actions",
+          style: { "flex-shrink": 1 },
+          cellType: "oxd-table-cell-actions",
+          cellConfig: {
+            delete: {
+              onClick: this.onClickDelete,
+              component: "oxd-icon-button",
+              props: {
+                name: "trash",
+              },
+            },
+            edit: {
+              onClick: this.onClickEdit,
+              props: {
+                name: "pencil-fill",
+              },
+            },
+          },
+        },
+      ],
+      items: [],
+      total: 0,
+      pages: 1,
+      currentPage: 1,
+      pageSize: 5,
+      showPaginator: false,
+      editItem: null,
+      checkedItems: [],
+    };
+  },
+
+  components: {
+    "delete-confirmation": DeleteConfirmationDialog,
+  },
+
+  watch: {
+    currentPage() {
+      this.resetDataTable();
+      this.fetchData();
+    },
+  },
+
+  created() {
+    this.fetchData();
+  },
+
+  methods: {
+    onClickAdd() {
+      this.$emit("onAddItem", { viewId:2, payload: null });
+    },
+    onClickEdit(item) {
+      this.$emit("onEditItem", { viewId:3, payload: item });
+    },
+    onClickDeleteSelected() {
+      const ids = this.checkedItems.map((index) => {
+        return this.items[index].id;
+      });
+      this.$refs.deleteDialog.showDialog().then((confirmation) => {
+        if (confirmation === "ok") {
+          this.deleteItems(ids);
+        }
+      });
+    },
+    onClickDelete(item) {
+      this.$refs.deleteDialog.showDialog().then((confirmation) => {
+        if (confirmation === "ok") {
+          this.deleteItems([item.id]);
+        }
+      });
+    },
+
+    fetchData() {
+      // TODO: Loading
+      let query = `limit=${this.pageSize}`;
```

Comment:
better to come with generic function add query param to API

Reply:
Implemented in `src/core/util/composable/usePaginate.ts`